File: app/src/main/java/me/sunhapper/spcharedittool/emoji/DeleteEmoji.java
package me.sunhapper.spcharedittool.emoji;

import android.content.Context;
import android.graphics.drawable.Drawable;
import me.sunhapper.spcharedittool.R;

public class DeleteEmoji implements Emoji {


  public DeleteEmoji() {

  }


  @Override
  public Drawable getDrawable(Context context) {
    return context.getResources().getDrawable(getDefaultResId());
  }

  public CharSequence getEmojiText() {
    return "";
  }

  @Override
  public Object getRes() {
    return R.drawable.common_emoj_delete_expression;
  }

  @Override
  public int getResType() {
    return 0;
  }

  @Override
  public boolean isDeleteIcon() {
    return true;
  }

  @Override
  public int getDefaultResId() {
    return R.drawable.common_emoj_delete_expression;
  }

  @Override
  public Object getCacheKey() {
    return R.drawable.common_emoj_delete_expression;
  }


}

File: app/src/main/java/me/sunhapper/spcharedittool/emoji/DeleteEmoji.kt
package me.sunhapper.spcharedittool.emoji

import android.content.Context
import android.graphics.drawable.Drawable
import android.support.v4.content.ContextCompat
import me.sunhapper.spcharedittool.R


class DeleteEmoji : Emoji {

    override val emojiText: CharSequence
        get() = ""

    override val res: Any
        get() = R.drawable.common_emoj_delete_expression


    override val isDeleteIcon: Boolean
        get() = true

    override val defaultResId: Int
        get() = R.drawable.common_emoj_delete_expression

    override val cacheKey: Any
        get() = R.drawable.common_emoj_delete_expression


    override fun getDrawable(context: Context): Drawable {
        return ContextCompat.getDrawable(context, defaultResId)!!
    }


}
